File: cli/src/command/template/typescript/src/vorpal.ts
import {
    ConfigContext,
    SYSTEMS,
    TypeScript,
    TypeScriptDevelopmentEnvironment,
} from "@altf4llc/vorpal-sdk";

const ctx = ConfigContext.create();

// -> 1. Activate: `source "$(vorpal build --path 'example-dev')/bin/activate"`
// -> 2. Deactivate: `deactivate`

await new TypeScriptDevelopmentEnvironment("example-dev", SYSTEMS)
    .build(ctx);

// -> 1. Build: `vorpal build 'example'`
// -> 2. Run: `$(vorpal build --path 'example')/bin/example`

await new TypeScript("example", SYSTEMS)
    .withEntrypoint("src/main.ts")
    .withIncludes(["src", "bun.lock", "package.json", "tsconfig.json"])
    .build(ctx);

await ctx.run();

File: cli/src/command/template/python/src/vorpal.py
from vorpal_sdk import SYSTEMS, ConfigContext, Python, PythonDevelopmentEnvironment

ctx = ConfigContext.create()

## -> 1. Activate: `source "$(vorpal build --path 'example-dev')/bin/activate"`
## -> 2. Deactivate: `deactivate`

(
    PythonDevelopmentEnvironment("example-dev", SYSTEMS)
    .build(ctx)
)

## -> 1. Build: `vorpal build 'example'`
## -> 2. Run: `$(vorpal build --path 'example')/bin/example`

(
    Python("example", SYSTEMS)
    .with_entrypoint("src/main.py")
    .with_includes(["pyproject.toml", "uv.lock", "src"])
    .build(ctx)
)

ctx.run()
